angular/playground/projects/ng-entity-service/src/lib/ng-entity.service.ts

```diff
@@ -0,0 +1,241 @@
+import { Observable, throwError } from 'rxjs';
+import { inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { catchError, finalize, map, tap } from 'rxjs/operators';
+import { HttpConfig, Msg, NgEntityServiceParams } from './types';
+import { EntityServiceAction, HttpMethod, NgEntityServiceNotifier } from './ng-entity-service-Notifier';
+import { NgEntityServiceLoader } from './ng-entity-service.loader';
+import { NG_ENTITY_SERVICE_CONFIG } from './ng-entity-service.config';
+import { isID } from './helpers';
+import { errorAction, successAction } from './action-factory';
+
+export const mapResponse = (config: HttpConfig) =>
+  map(res => ((config || {}).mapResponseFn ? config.mapResponseFn(res) : res));
+
+export class NgEntityService<S extends EntityState = any> extends EntityService<S> {
+  private http: HttpClient;
+  private notifier: NgEntityServiceNotifier;
+  private globalConfig: NgEntityServiceParams = {};
+  private mergedConfig: NgEntityServiceParams;
+  private loader: NgEntityServiceLoader;
+
+  private dispatchSuccess: (action: Partial<EntityServiceAction>) => void;
+  private dispatchError: (action: Partial<EntityServiceAction>) => void;
+
+  constructor(protected store: EntityStore<S>, private config: NgEntityServiceParams = {}) {
+    super();
+    this.http = inject(HttpClient);
+    this.loader = inject(NgEntityServiceLoader);
+    this.notifier = inject(NgEntityServiceNotifier);
+    this.globalConfig = inject(NG_ENTITY_SERVICE_CONFIG);
+
+    this.mergedConfig = { ...this.globalConfig, ...config };
+
+    this.dispatchSuccess = successAction(this.store.storeName, this.notifier);
+    this.dispatchError = errorAction(this.store.storeName, this.notifier);
+  }
+
+  get api() {
+    return `${this.getConfigValue('baseUrl')}/${this.resourceName}`;
+  }
+
+  get resourceName() {
+    return this.getConfigValue('resourceName') || this.store.storeName;
+  }
+
+  /**
+   *
+   * Get all or one entity - Creates a GET request
+   *
+   * service.get().subscribe()
+   * service.get({ headers, params, url })
+   *
+   * service.get(id)
+   * service.get(id, { headers, params, url })
+   *
+   */
+  get<T>(id?: getIDType<S>, config?: HttpConfig & { append?: boolean } & Msg): Observable<T>;
+  get<T>(config?: HttpConfig & { append?: boolean } & Msg): Observable<T>;
+  get<T>(idOrConfig?: getIDType<S> | HttpConfig, config?: HttpConfig & { append?: boolean } & Msg): Observable<T> {
+    let url: string;
+    const isSingle = isID(idOrConfig);
+    const _config: HttpConfig & { append?: boolean } & Msg = (isSingle ? config : idOrConfig) || {};
+
+    if (_config.url) {
+      url = _config.url;
+    } else {
+      url = isSingle ? `${this.api}/${idOrConfig}` : this.api;
+    }
+
+    this.loader.dispatch({
+      method: HttpMethod.GET,
+      loading: true,
+      entityId: isSingle ? idOrConfig : null,
+      storeName: this.store.storeName
+    });
+
+    return this.http.get(url).pipe(
```

Comment:
Should you not put the options as second parameter?

Reply:
Also, maybe check somewhere that responseType isn't blob, arrayBuffer, ... to warn the dev that this is not supported.

Reply:
Yes, thanks, missed it. They can't change the `responseType`, and it will be in the docs, so I think it's redundant for now.